### PR TITLE
Release 1.4.4 update from upstream

### PR DESCRIPTION
**Pull request by iramiller**
Maybe fixes to gossip issues?  Looks like a lot of tickets related to cert updates and broken gossip like we are experiencing in test.

### CODE REVIEW

File: common/crypto/expiration.go [hunk 1]
```diff
@@ -34,3 +38,60 @@ func ExpiresAt(identityBytes []byte) time.Time {
...
 }
+
+// WarnFunc notifies a warning happened with the given format, and can be replaced with Warnf of a logger.
+type WarnFunc func(format string, args ...interface{})
+
+// Scheduler invokes f after d time, and can be replaced with time.AfterFunc.
+type Scheduler func(d time.Duration, f func()) *time.Timer
+
+// TrackExpiration warns a week before one of the certificates expires
+func TrackExpiration(tls bool, serverCert []byte, clientCertChain [][]byte, sIDBytes []byte, warn WarnFunc, now time.Time, s Scheduler) {
```
Comment (iramiller) [inline common/crypto/expiration.go:49]: This is an interesting section to see come into the code base...

Comment (arnabmitra) [reply common/crypto/expiration.go:49]: nice

**Review by arnabmitra — APPROVED** [review]

File: ci/install_deps.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ steps:
       sudo apt-get clean
       sudo apt-get update
       sudo apt-get install -y libtool gcc make
+      echo "vsts  hard  nofile  65535" | sudo tee -a /etc/security/limits.conf
+      echo "vsts  soft  nofile  65535" | sudo tee -a /etc/security/limits.conf
```
Comment (iramiller) [inline ci/install_deps.yml:11]: :hmm:

File: gossip/comm/comm_impl.go [hunk 3]
```diff
@@ -232,6 +237,7 @@ func (c *commImpl) sendToEndpoint(peer *RemotePeer, msg *proto.SignedGossipMessa
 		disConnectOnErr := func(err error) {
 			c.logger.Warningf("%v isn't responsive: %v", peer, err)
 			c.disconnect(peer.PKIID)
+			conn.close()
```
Comment (ghost) [inline gossip/comm/comm_impl.go:240]: ABC, always be closing.

**Review by ghost — APPROVED** [review]
Should be fine

**Pull request merged**